src/ResultsActivity/ResultsActivityContainer.js

```diff
@@ -27,31 +30,24 @@ class ResultsActivityContainer extends Component {
     this.handlePrint = this.handlePrint.bind(this)
     this.setColumnSort = this.setColumnSort.bind(this)
     this.getActivityList = this.getActivityList.bind(this)
-    this.update = this.update.bind(this)
+    this.onListCriteriaChange = this.onListCriteriaChange.bind(this)
+    this.reload = this.reload.bind(this)
     this.state = {
       redactedPrint: false,
     }
   }
 
   async componentDidMount() {
-    const { activity, history } = this.props
+    const { location: routerLocation, getAbsentReasonsDispatch } = this.props
+    const { search } = routerLocation
 
-    try {
-      if (activity) {
-        this.getActivityList()
-      } else {
-        history.push('/manage-prisoner-whereabouts')
-      }
-    } catch (error) {
-      this.handleError(error)
-    }
-  }
+    const { date, period, location } = queryString.parse(search)
 
-  async componentDidUpdate(prevProps) {
-    const { date, period } = this.props
-
-    if ((prevProps.date && prevProps.date !== date) || (prevProps.period && prevProps.period !== period)) {
-      await this.update()
+    if (!date && !period && !location) {
+      window.location = '/manage-prisoner-whereabouts'
+    } else {
+      getAbsentReasonsDispatch()
```

Comment:
Pulled the absent reason request up into componentDidMount so that it only gets called once.